SharedCryptoPkg/Include/Private/Shared/SharedCrypto.h

```diff
@@ -0,0 +1,1203 @@
+//    Symmetric Cryptography Primitive
+//=====================================================================================
+
+typedef
+UINTN
+(EFIAPI *SHAREDCRYPTO_TDES_GetContextSize) (
+  VOID
+  );
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_TDES_Init) (
+  OUT  VOID         *TdesContext,
+  IN   CONST UINT8  *Key,
+  IN   UINTN        KeyLength
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_TDES_EcbEncrypt) (
+  IN   VOID         *TdesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  OUT  UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_TDES_EcbDecrypt) (
+  IN   VOID         *TdesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  OUT  UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_TDES_CbcEncrypt) (
+  IN   VOID         *TdesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  IN   CONST UINT8  *Ivec,
+  OUT  UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_TDES_CbcDecrypt) (
+  IN   VOID         *TdesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  IN   CONST UINT8  *Ivec,
+  OUT  UINT8        *Output
+  );
+
+typedef
+UINTN
+(EFIAPI *SHAREDCRYPTO_AES_GetContextSize) (
+  VOID
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_AES_Init) (
+  OUT  VOID         *AesContext,
+  IN   CONST UINT8  *Key,
+  IN   UINTN        KeyLength
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_AES_EcbEncrypt) (
+  IN   VOID         *AesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  OUT  UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_AES_EcbDecrypt) (
+  IN   VOID         *AesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  OUT  UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_AES_CbcEncrypt) (
+  IN   VOID         *AesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  IN   CONST UINT8  *Ivec,
+  OUT  UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_AES_CbcDecrypt) (
+  IN   VOID         *AesContext,
+  IN   CONST UINT8  *Input,
+  IN   UINTN        InputSize,
+  IN   CONST UINT8  *Ivec,
+  OUT  UINT8        *Output
+  );
+
+typedef
+UINTN
+(EFIAPI *SHAREDCRYPTO_ARC4_GetContextSize) (
+  VOID
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_ARC4_Init) (
+  OUT  VOID         *Arc4Context,
+  IN   CONST UINT8  *Key,
+  IN   UINTN        KeySize
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_ARC4_Encrypt) (
+  IN OUT  VOID         *Arc4Context,
+  IN      CONST UINT8  *Input,
+  IN      UINTN        InputSize,
+  OUT     UINT8        *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_ARC4_Decrypt) (
+  IN OUT  VOID   *Arc4Context,
+  IN      UINT8  *Input,
+  IN      UINTN  InputSize,
+  OUT     UINT8  *Output
+  );
+
+
+typedef
+BOOLEAN
+(EFIAPI *SHAREDCRYPTO_ARC4_Reset) (
+  IN OUT  VOID  *Arc4Context
+  );
+
+#pragma pack(push, 1)
```

Comment:
Should not be any reason to use #pragma pack for a PPI, Protocol, SMM Protocol.

Reply:
We were having some trouble without this (strange memory errors and page faults in PEI and SMM). It could be possible that this is no longer needed. I'll test without it tomorrow on a device to make sure it can boot without it.

Reply:
Alright removed the pragma pack and it looks like it's still working